packages/airnode-utilities/src/caching/index.ts

```diff
@@ -0,0 +1,252 @@
+import { existsSync, mkdirSync, readdirSync, readFileSync, rmdirSync, rmSync, statSync, writeFileSync } from 'fs';
+import { spawn } from 'child_process';
+import { join } from 'path';
+import { logger } from '../logging';
+
+export const CACHE_BASE_PATH = `/tmp/airnode-cache`;
+export const CACHE_MAX_FILESYSTEM_AGE_MINUTES = 60;
+export const CACHE_MAX_FILES = 100_000; // (500 (cache size) * 1024 (KB) ) / 4 (KB, fs "cluster size" = 130k
+
+export const isJest = () => process.env.JEST_WORKER_ID !== undefined;
+
+/**
+ * Calls the OS sync command to forcefully sync the cache filesystem.
+ * This is the functional equivalent of "commit" from SQL.
+ */
+const syncFsSync = () => {
+  try {
+    // TODO find a way to mock out spawn...
+    if (!isJest()) {
+      spawn(`sync`, [`-f`, CACHE_BASE_PATH]);
+    }
+  } catch (e) {
+    logger.error(`Unable to sync cache fs`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Syncs the temporary filesystem asynchronously.
+ * Lambda will "freeze" and "thaw" the execution context and with it any async promises in progress. We create and
+ * forget this promise here, but it will resolve at some point, either in this invocation or a later one.
+ */
+const syncFsASync = () => {
+  try {
+    new Promise(() => {
+      spawn(`sync`, [`-f`, CACHE_BASE_PATH]);
+    });
+  } catch (e) {
+    logger.error(`Unable to sync cache fs`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Initialise the path for the fs cache
+ */
+const initPath = () => {
+  try {
+    if (!existsSync(CACHE_BASE_PATH)) {
+      mkdirSync(CACHE_BASE_PATH);
+      syncFsSync();
+    }
+  } catch (e) {
+    logger.error(`Unable to init fs cache`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Get all keys from the cache that start with startingKey.
+ *
+ * @param startingKey should take the form of `blockedRequestId-` and the full key could be `blockedRequestId-0x0000000...`
+ */
+const getKeys = (startingKey?: string): string[] => {
+  try {
+    const keys = readdirSync(CACHE_BASE_PATH);
+    if (startingKey) {
+      return keys.filter((key) => key.indexOf(startingKey) === 0);
```

Comment:
```suggestion
      return keys.filter((key) => key.startsWith(startingKey) === 0);
```

because the function is misleading currently.

Reply:
Nice :)